server/supernova/server/server.cpp

```diff
@@ -210,7 +211,13 @@ void nova_server::rebuild_dsp_queue(void) {
 static void name_current_thread(int thread_index) {
     char buf[1024];
     sprintf(buf, "DSP Thread %d", thread_index);
+#if defined(__linux__)
     name_thread(buf);
+#elif defined(__APPLE__)
+    // TODO
```

Comment:
Maybe there's a way to do this in macOS, I haven't looked (yet). This is mainly a Windows PR :-)

`name_thread` gives a name to the thread which can be shown by debuggers. This is handy for debugging supernova because it will show you which DSP thread you're in.

Reply:
A little bit of googling pointed me to `pthread_setname_np`. It is supposed to work on macOS. Will give it a try.

On Linux it's just a wrapper around `prctl(PR_SET_NAME, th_name)`, i.e. the same as in `nova::name_thread`.

It is also available in winpthreads, but it uses the "old" exception based method described in https://docs.microsoft.com/en-us/visualstudio/debugger/how-to-set-a-thread-name-in-native-code?view=vs-2022. See https://github.com/mirror/mingw-w64/blob/b628bd605dea2b51f0493689b4889452b369e014/mingw-w64-libraries/winpthreads/src/thread.c#L69. I could still use it as a fallback for Windows 7 (where `win32_name_thread` currently does nothing).

Reply:
Thanks! So this is just used for debugging, not for actual threading functionality, correct?

Reply:
I think it will also show up in core dumps. But yes, it doesn't have any functionality.

Reply:
Sounds good! If you want to look at the macOS and older Windows implementation that's great, but totally optional for this PR. Thanks again for the explanation!

Reply:
I'll do it. I'm quite busy at the moment, but I will continue in a few days.